src/rust/engine/src/nodes.rs

```diff
@@ -542,28 +511,68 @@ impl SelectDependencies {
 #[derive(Clone, Debug, Eq, Hash, PartialEq)]
 pub struct ExecuteProcess(process_executor::ExecuteProcessRequest);
 
+impl ExecuteProcess {
+  ///
+  /// Lifts a Key representing a python ExecuteProcessRequest value into a ExecuteProcess Node.
+  ///
+  fn lift(subject: &Key) -> ExecuteProcess {
+    let value = externs::val_for(subject);
+
+    let mut env: BTreeMap<String, String> = BTreeMap::new();
+    let env_var_parts = externs::project_multi_strs(&value, "env");
+    // TODO: Error if env_var_parts.len() % 2 != 0
+    for i in 0..(env_var_parts.len() / 2) {
+      env.insert(
+        env_var_parts[2 * i].clone(),
+        env_var_parts[2 * i + 1].clone(),
+      );
+    }
+
+    // TODO: Make this much less unwrap-happy with https://github.com/pantsbuild/pants/issues/5502
+    let fingerprint = externs::project_str(&value, "input_files_digest");
+    let digest_length = externs::project_str(&value, "digest_length");
+    let digest_length_as_usize = digest_length.parse::<usize>().unwrap();
+    let digest = hashing::Digest(
+      hashing::Fingerprint::from_hex_string(&fingerprint).unwrap(),
+      digest_length_as_usize,
+    );
+
+    ExecuteProcess(process_executor::ExecuteProcessRequest {
+      argv: externs::project_multi_strs(&value, "argv"),
+      env: env,
+      input_files: digest,
+    })
+  }
+}
+
 #[derive(Clone, Debug)]
 pub struct ProcessResult(process_executor::ExecuteProcessResult);
 
 impl Node for ExecuteProcess {
   type Output = ProcessResult;
 
   fn run(self, context: Context) -> NodeFuture<ProcessResult> {
-    let request = self.0.clone();
-
-    // TODO: Make this much less unwrap-happy with https://github.com/pantsbuild/pants/issues/5502
-
-    let tmpdir = TempDir::new("process-execution").unwrap();
+    let request = self.0;
+    let context2 = context.clone();
+    // TODO: Process pool management should likely move into the `process_executor` crate, which
```

Comment:
`process_execution` crate, actually; `process_executor` is the standalone binary which consumes `process_execution` :)

(I have no idea why `process_execution` is used as `process_executor` in this file. It shouldn't be :)